nms: add page (#5483)
tldr-pages/tldr: pages/common/nms.md

# nms

> A cool (piped) text decryption effect.
> More information: <https://github.com/bartobri/no-more-secrets>.

- Decrypt "Hello, World!" after a keystroke:

`echo "Hello, World! | nms"`

> - Decrypt the output of `ls -la` automatically:

Reviewer: This sort of 'automatic' behavior, while maybe not default to this command, is generally expected from most commands, so 'automatically' is more likely to confuse the user.
```suggestion
- Decrypt the output of `ls -la`:
```

Author: I agree with you that people usually expect this behavior by default, and for instance in the next examples I think you're right about ignoring "automatically" in their description. However, if in the first example "nms" is used, and in this one "nms -a" is used, I would say people will be more confused if not explained what's the difference.

Reviewer: Ah, good point.  In that case, what do you think showing the 'automatic' behavior first and the delayed behavior later?

Author: Honestly, I think we should leave it as is. It is a problem of the command itself, as you say the default behavior should be automatically taking place, with an option to make it non automatic, however this is reversed, leading to confusion.

IMO there's no simple non-confusing way to fix it here, if we do as you say (first showcasing the automatic behavior, then not) it will a bit awkward. Still, if you insist I have no problem changing it.

It would be nice to hear other opinions as well.

Reviewer: >It would be nice to hear other opinions as well.

I agree.  Applying the other suggestions would make this one much more noticeable ;)

> `ls -la | nms -a`

Reviewer: ```suggestion
`{{ls -la}} | nms -a`
```

> - Decrypt output, automatically:

Reviewer 2: Hey there - welcome to tldr-pages! Great first page here. Reading the README for the project behind this command, the `automatically` does seem to hold meaning here. What about this instead to aid clarity?

```suggestion
- Decrypt output immediately, without waiting for a keystroke:
```

Author: Hi @sbrl, thanks for the welcome! I like your suggestion quite a lot, what about you @bl-ue? Would it settle the debate?

Reviewer 3: Yeah, sounds like the user doesn't need to trigger the animation with `space`, or any key. Makes sense.

Reviewer: +1 for @sbrl's suggestion—kills all the birds with one stone :D

> - Decrypt the content of `secret_message.txt` automatically with red output:

Reviewer: We try to be generic when possible, using simply 'a file' when it will work just as well as an actual file name.
```suggestion
- Decrypt the content of a file automatically with a custom output color:
```

> `cat secret_message.txt | nms -a -f red`

Reviewer: If the list is short enough, include the whole list of valid values separated by `|` (I put the default `blue` first, otherwise thie order is that found in the man page):
```suggestion
`cat {{path/to/file}} | nms -a -f {{blue|white|yellow|black|magenta|green|red}}`
```

> - Clear the screen prior to decrypt `git status` automatically:
> 
> `git status | nms -a -c`

Reviewer: I think it would be better to just use 'command' here
```suggestion
- Clear the screen before decrypting:

`{{command}} | nms -a -c`
```
